### ISSUE TITLE
Any plans to add support of NewtonCosine and NewtonL2 score_functions when training on CPU ?

### ISSUE THREAD

**Issue by Meleeri** [catboost/catboost#2290]
I'm training a binary classification model on a dataset which comprised of roughly several millon samples, each containing tens of thousands of numerical features, relatively sparse though. After few hours of parameter tuning, I still got worse AUC metric than results trained on the same dataset using xgboost. Since xgboost used taylor expansion for calculating score function, I thought lacking support of NewtonCosine and NewtonL2 score_functions on CPU might be the cause of degraded performance using catboost. I really like catboost's automatic feature selection and categorical feature support , and observed less overfitting on the training set compared to xgboost, but due to hardware limitations I'm unable to train models on GPU. So are there any plans to add support of those?

**Comment by bheeso** [comment]
I also hope that NewtonCosine and NewtonL2 score_functions can be supported on CPU.
